chanzuckerberg/single-cell-data-portal
# feat(doi update): add retry to dataset_metadata_update

## Reason for Change

- On rare occasion AWS has failed to start a batch job with "Task failed to start" which can be attributed to transient issues with AWS. To avoid these problems when updating a DOI or other metadata fields, a retry as been added to the batch job. Any other error will result in the batch job exiting.

## Changes

- retry "Task failed to start" error for batch job
- attempt to retry 3 times.
- exit batch job for all other errors.

## Testing steps

- updated DOI and verify everything passes.
- I welcome suggestions on forcing a "Task failed to start" error.
